### PR TITLE
Migrate designs to rtk

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

This PR fixes #

- [x] move patterns fetching to rtk 
- [x] wrapping meshery patterns component to redux-provider wrapper 
- [x] move provider capabilities to rtk query
- [x] cloning, download, deleting of designs
- [x] move import and publish api to rtk
- [ ] remove schema fetching and use sistent schemas for import and publish of designs

> [!NOTE]
> Each tasks above are independent tasks so they are not dependent on each other, even if we want to merge we can because this will not break the ui.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sudhanshutech** [comment]
@dottharun it will be nice if checkout this branch and test , thanks.

File: ui/components/MesheryPatterns.js [hunk 1]
```diff
@@ -1824,6 +1769,14 @@ const PublishModal = React.memo((props) => {
   );
 });
 
+const MesheryPatternsProvider = (props) => {
+  return (
```
Comment (Aut0R3V) [inline ui/components/MesheryPatterns.js:1773]: I think the provided inputs solve the problem to some extent

**Comment by sudhanshutech** [comment]
The left task of schema to use sistent schema can be done in iterative pr because that will not include rtk migration just use schema and fix around that.

**Review by dottharun — CHANGES_REQUESTED** [review]

File: ui/components/MesheryPatterns.js [hunk 2]
```diff
@@ -1824,6 +1769,14 @@ const PublishModal = React.memo((props) => {
   );
 });
 
+const MesheryPatternsProvider = (props) => {
```
[the remark this answers is not in the document]
Comment (sudhanshutech) [reply ui/components/MesheryPatterns.js]: fixed

File: ui/components/MesheryPatterns.js [hunk 3]
```diff
@@ -1535,7 +1480,7 @@ function MesheryPatterns({
           message: `"${name}" design uploaded`,
           event_type: EVENT_TYPES.SUCCESS,
         });
-        fetchPatternsCaller()();
+        // fetchPatternsCaller()();
```
[the remark this answers is not in the document]
Comment (sudhanshutech) [reply ui/components/MesheryPatterns.js]: fixed

**Comment by sudhanshutech** [comment]
All feedbacks incorporated @dottharun , thanks

File: ui/components/MesheryPatterns.js [hunk 4]
```diff
@@ -1485,9 +1430,9 @@ function MesheryPatterns({
     },
   };
 
-  if (loading) {
-    return <LoadingScreen animatedIcon="AnimatedMeshPattern" message={`Loading ${pageTitle}...`} />;
-  }
+  // if (loading) {
```
[the remark this answers is not in the document]
Comment (sudhanshutech) [reply ui/components/MesheryPatterns.js]: fixed

**Pull request merged**